docs/source-2.0/guides/generating-code/update-model.rst

```diff
@@ -0,0 +1,140 @@
+``build.gradle.kts``:
+
+.. code-block:: kotlin
+
+    dependencies {
+        ...
+        implementation("software.amazon.smithy:smithy-aws-traits:__smithy_version__")
+    }
+
+Now, import the :ref:`@aws.protocols#restJson1 <aws.protocols#restJson1-trait>` trait
+and apply it to the ``Weather`` service shape:
+
+.. code-block:: smithy
+
+    $version: "2"
+    namespace example.weather
+
+    use aws.protocols#restJson1
+
+    /// Provides weather forecasts.
+    @paginated(
+        inputToken: "nextToken"
+        outputToken: "nextToken"
+        pageSize: "pageSize"
+    )
+    @restJson1
+    service Weather {
+        version: "2006-03-01"
+        resources: [City]
+        operations: [GetCurrentTime]
+    }
+
+Adding HTTP bindings
+====================
+
+In Smithy, HTTP can be configured by applying :ref:`HTTP binding traits <http-traits>`
+to operation shapes. HTTP protocols can use these traits to generate code that formats
+HTTP messages properly.
+
+
+First, configure the HTTP method, request URI, and the status code of a successful
+response with the :ref:`@http trait <http-trait>`.
+
+.. code-block:: smithy
+
+    @readonly
+    @http(code: 200, method: "GET", uri: "/cities/{cityId}")
+    operation GetCity {
+        input: GetCityInput
+        output: GetCityOutput
+        errors: [NoSuchResource]
+    }
+
+    @paginated(items: "items")
+    @readonly
+    @http(code: 200, method: "GET", uri: "/cities")
+    operation ListCities {
+        input: ListCitiesInput
+        output: ListCitiesOutput
+    }
+
+    @readonly
+    @http(code: 200, method: "GET", uri: "/currentTime")
+    operation GetCurrentTime {
+        input: GetCurrentTimeInput
+        output: GetCurrentTimeOutput
+    }
+
+    @readonly
+    @http(code: 200, method: "GET", uri: "/forecast/{cityId}")
+    operation GetForecast {
+        input: GetForecastInput
+        output: GetForecastOutput
+    }
+
+The URI patterns for the ``GetCity`` and ``GetForecast`` operations each use an HTTP label to
+bind the ``cityId`` member of the operation input structure to the request URI. Let's specify
+the members that should be bound to the URIs using the :ref:`@httpLabel trait <httplabel-trait>`:
+
+
+.. code-block:: smithy
+
+    @input
+    structure GetCityInput {
+        // "cityId" provides the identifier for the resource and
+        // has to be marked as required.
+        @required
+        @httpLabel
+        cityId: CityId
+    }
+
+    @input
+    structure GetForecastInput {
+        @required
+        @httpLabel
+        cityId: CityId
+    }
+
+For the ``ListCities`` operation, include the ``nextToken`` and ``pageSize`` input members
+in the request URI as query parameters using the :ref:`@httpQuery trait <httpquery-trait>`:
+
+.. code-block:: smithy
+
+    @input
+    structure ListCitiesInput {
+        @httpQuery("nextToken")
+        nextToken: String
+        @httpQuery("pageSize")
+        pageSize: Integer
+    }
```

Comment:
I mentioned it offline, but the intention was to add a directory with the complete example within the repo, I just wasn't sure how exactly I wanted to do it. There are a lot of files in this example, and I think it would be easier for a user to just look at that. Plus easier for us to maintain.